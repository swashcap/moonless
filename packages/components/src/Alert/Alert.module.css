.Alert {
  border-radius: var(--component-alert-border-radius);
  border-style: solid;
  border-width: var(--component-alert-border-width);
  padding: var(--component-alert-padding-vertical) var(--component-alert-padding-horizontal);
}

.AlertOnClose {
  position: relative;
}

.AlertButton {
  -webkit-appearance: none;
  background: none;
  border: none;
  color: inherit;
  cursor: pointer;
  line-height: 1;
  margin: 0;
  padding: 0.5rem;
  position: absolute;
  right: 0;
  top: 0;
}

/** Variant */

.AlertError {
  background: var(--component-alert-error-background-color);
  border-color: var(--component-alert-error-border-color);
  color: var(--component-alert-error-text-color);
}

.AlertSuccess {
  background: var(--component-alert-success-background-color);
  border-color: var(--component-alert-success-border-color);
  color: var(--component-alert-success-text-color);
}

.AlertWarning {
  background: var(--component-alert-warning-background-color);
  border-color: var(--component-alert-warning-border-color);
  color: var(--component-alert-warning-text-color);
}
