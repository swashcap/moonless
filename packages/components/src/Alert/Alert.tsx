import React from 'react';
import { clsx } from 'moonless-utils';

import { XIcon } from '../Icon/Icon';

import styles from './Alert.module.css';

export type AlertVariant = 'error' | 'success' | 'warning';

export interface AlertProps
  extends React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  > {
  onClose?: React.DOMAttributes<HTMLButtonElement>['onClick'];
  variant?: AlertVariant;
}

export class Alert extends React.Component<AlertProps> {
  private buttonRef = React.createRef<HTMLButtonElement>();

  focusButton() {
    this.buttonRef.current?.focus();
  }

  render() {
    const {
      children,
      className,
      onClose,
      variant = 'warning',
      ...rest
    } = this.props;

    return (
      <div
        className={clsx(
          styles.Alert,
          variant === 'error' && styles.AlertError,
          variant === 'success' && styles.AlertSuccess,
          variant === 'warning' && styles.AlertWarning,
          onClose && styles.AlertOnClose,
          className
        )}
        role="alert"
        {...rest}
      >
        {children}
        {!!onClose && (
          <button
            aria-label="Close"
            className={styles.AlertButton}
            onClick={onClose}
            ref={this.buttonRef}
          >
            <XIcon />
          </button>
        )}
      </div>
    );
  }
}
