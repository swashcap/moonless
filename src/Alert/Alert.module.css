.Alert {
  align-items: center;
  display: flex;
  justify-content: space-between;
  min-height: 2.5rem;
  padding: 0 1rem;
}

.AlertButton {
  background: none;
  border: none;
  color: inherit;
  cursor: pointer;
  margin: 0 -1rem 0 0;
  padding: 0.5rem;
  vertical-align: top;
}

/** Variant */

.AlertError {
  background: var(--color-red);
  color: var(--color-white);
}

.AlertSuccess {
  background: var(--color-green);
  color: var(--color-white);
}

.AlertWarning {
  background: var(--color-yellow);
  color: var(--color-black);
}
