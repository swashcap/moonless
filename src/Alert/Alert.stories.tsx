import React from 'react';

import { Alert } from './Alert';

export default {
  component: Alert,
  title: 'Alert',
};

export const Default = () => <Alert>Default</Alert>;

export const OnClose = () => <Alert onClose={console.log}>On Close</Alert>;

export const Variant = () => (
  <>
    <Alert variant="error">Error</Alert>
    <Alert variant="success">Success</Alert>
    <Alert variant="warning">Warning</Alert>
  </>
);
