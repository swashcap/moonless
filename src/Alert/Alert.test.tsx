import React from 'react';
import { render } from '@testing-library/react';

import { Alert } from './Alert';

describe('Alert', () => {
  test('It should render.', () => {
    const { container } = render(<Alert>Alert</Alert>);

    expect(container).toMatchSnapshot();
  });
});
