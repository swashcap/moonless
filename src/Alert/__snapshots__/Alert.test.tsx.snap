// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert It should render. 1`] = `
<div>
  <div
    class="Alert AlertWarning"
  >
    Alert
  </div>
</div>
`;
